micrometer-observation/build.gradle

```diff
@@ -39,19 +39,15 @@ jar {
 dependencies {
     api project(":micrometer-commons")
 
-    optionalApi 'io.micrometer:context-propagation'
+    optionalApi libs.contextPropagation
 
     // HttpServlet KeyValueProvider
-    optionalApi 'javax.servlet:javax.servlet-api'
+    optionalApi libs.javax.servletApi
 
     // Aspects
     optionalApi libs.aspectjrt
 
-    // log monitoring
-    testImplementation 'ch.qos.logback:logback-classic'
-    testImplementation 'org.apache.logging.log4j:log4j-core'
-
-    testImplementation project(':micrometer-observation-test')
+    testRuntimeOnly libs.logback12
```

Comment:
This was perhaps only needed because archunit seems to have a runtime dependency on slf4j. I didn't look into it too deeply. But log4j was not used or needed at all in this project so I got rid of that dependency while reducing the scope of the logback dependency.